proto/hummock.proto

```diff
@@ -312,6 +337,7 @@ service HummockManagerService {
   rpc UnpinVersion(UnpinVersionRequest) returns (UnpinVersionResponse);
   rpc UnpinVersionBefore(UnpinVersionBeforeRequest) returns (UnpinVersionBeforeResponse);
   rpc ReportCompactionTasks(ReportCompactionTasksRequest) returns (ReportCompactionTasksResponse);
+  rpc ReportCompactionTaskProgress(ReportCompactionTaskProgressRequest) returns (ReportCompactionTaskProgressResponse);
```

Comment:
I think we can just make SubscribeCompactTasks rpc to be a bidi-streaming rpc and use this streaming rpc as the only compaction related communication channel so that we don't need to create a dedicated unary rpc for CompactTaskProgress reporting.

Reply:
I don't quite get the benefit of unifying all RPCs into a stream, because we still have to dispatch messages based on their type if using stream?

Reply:
> we still have to dispatch messages based on their type if using stream

True. The dispatch logic is simple though. IMO, the main benefit is to simplify failure handling and retry:
- We have one unify logic to handle rpc failure, which is to re-create the stream. We don't need to consider the case when we have one rpc fails but the other rpc still works.
- We don't need to worry about timeout / retry since the streaming rpc has its own mechanism to ensure delivery.

The side benefits are:
- Reduce the number of messages transmitted because we don't need to explicitly ack each request of ReportCompactionTask / ReportCompactionTaskProgress.
- Reduce overhead of rpc creation/destruction. In gRPC, some application-level metadata needs to be transmitted between two ends to initiate/finish a rpc and we can avoid these work by have one streaming rpc instead of many unary rpcs.

(technially these side benefits are the benefits of using streaming rpc, not unifying all rpcs into one streaming rpc)

Reply:
I tried implementing as a stream, it was quite troublesome to manage incoming streams. We need to add a task to consume the stream rather than allow new RPC to trigger a task. 

Anyway, maybe this is incremental work that can be reserved for next PR if it is deemed necessary. 

Basically, we should just have a global stream, and each compactor added to meta will pipe the input stream into the global stream to be consumed by the global "update compactor progress" task.

Reply:
Anw, this fits the mpsc pattern, we can actually clone the tx for each compactor. I'm just not sure how we'd manage the lifetime.